docs/source/pytest.rst

```diff
@@ -0,0 +1,489 @@
+* filtering cocotb tests with `pytest`_ ``-k '<expression>'`` and ``-m '<markers>'`` options.
+* reporting all executed cocotb tests.
+* parallel execution of cocotb runners by using the `pytest-xdist`_ plugin
+
+Enabling the Plugin
+===================
+
+:py:mod:`cocotb_tools.pytest.plugin` can be enabled in various ways.
+
+In a Python project
+-------------------
+
+When using the `pyproject.toml`_ file (recommended way):
+
+.. code:: toml
+
+    [project.entry-points.pytest11]
+    cocotb = "cocotb_tools.pytest.plugin"
+
+When using the ``pytest.ini`` file:
+
+.. code:: ini
+
+    [pytest]
+    addopts = -p cocotb_tools.pytest.plugin
+
+When using the ``setup.cfg`` file:
+
+.. code:: ini
+
+    [options.entry_points]
+    pytest11 =
+      cocotb = cocotb_tools.pytest.plugin
+
+When using the ``setup.py`` file:
+
+.. code:: python
+
+    from setuptools import setup
+
+    setup(
+        # ...,
+        entry_points={
+            "pytest11": [
+                "cocotb = cocotb_tools.pytest.plugin",
+            ],
+        },
+    )
+
+In a non-Python project
+-----------------------
+
+By defining the global variable ``pytest_plugins`` when using a ``conftest.py`` file
+(which must be located in the root of the project):
+
+.. code:: python
+
+    pytest_plugins = ("cocotb_tools.pytest.plugin",)
+
+By defining the ``PYTEST_PLUGINS`` environment variable:
+
+.. code:: shell
+
+    export PYTEST_PLUGINS="cocotb_tools.pytest.plugin"
+
+By using the ``-p <plugin>`` option when invoking the `pytest`_ command line interface:
+
+.. code:: shell
+
+    pytest -p cocotb_tools.pytest.plugin ...
+
+Building and Testing
+====================
+
+:py:class:`cocotb_tools.pytest.hdl.HDL` interfaces with the :ref:`Python runners <howto-python-runner>` to build designs and run simulations.
+The :py:class:`~cocotb_tools.runner.Runner` is fully configurable by using ``--cocotb-*`` command line arguments,
+configuration files like `pyproject.toml`_ or `fixture`_ arguments.
+
+The plugin provides an :fixture:`~cocotb_tools.pytest.plugin.hdl` fixture that will create a new instance of
+:py:class:`~cocotb_tools.pytest.hdl.HDL` that can be customized and then used in tests.
+
+An example is provided below, located in a project ``conftest.py`` file:
+
+.. code:: python
+
+    import pytest
+    from cocotb_tools.pytest.hdl import HDL
+
+
+    @pytest.fixture(name="sample_module")
+    def sample_module_fixture(hdl: HDL) -> HDL:
+        """Define HDL design by adding HDL source files to it.
+
+        Args:
+            hdl: Fixture created by the cocotb pytest plugin, representing a HDL design.
+
+        Returns:
+            Representation of HDL design with added HDL source files.
+        """
+        hdl.sources = (
+            # List HDL source files,
+            "sample_module.sv",
+        )
+
+        # Build HDL design
+        hdl.build()
+
+        return hdl
+
+
+.. currentmodule:: cocotb_tools
```

Comment:
Avoid currentmodule.

Reply:
I think I'm going back on this. Lets just leave it be.

Reply:
@ktbarrett I have completely removed `.. currentmodule:: cocotb_tools` in this commit e7d863210679ba72a7315db0873d14c3193845fd